app/models/display_ad.rb

```diff
@@ -35,32 +35,8 @@ class DisplayAd < ApplicationRecord
                      }
 
   def self.for_display(area, user_signed_in, article_tags = [])
-    relation = approved_and_published.where(placement_area: area).order(success_rate: :desc)
-
-    if article_tags.any?
-      display_ads_with_no_tags = relation.where(cached_tag_list: "")
-      display_ads_with_targeted_article_tags = relation.cached_tagged_with_any(article_tags)
-
-      relation = display_ads_with_no_tags.or(display_ads_with_targeted_article_tags)
-    end
-
-    if article_tags.blank?
-      relation = relation.where(cached_tag_list: "")
-    end
-
-    relation = if user_signed_in
-                 relation.where(display_to: %w[all logged_in])
-               else
-                 relation.where(display_to: %w[all logged_out])
-               end
-
-    relation.order(success_rate: :desc)
-
-    if rand(8) == 1
-      relation.sample
-    else
-      relation.limit(rand(1..15)).sample
-    end
+    DisplayAds::FilteredAdsQuery.call(display_ads: self, area: area, user_signed_in: user_signed_in,
+                                      article_tags: article_tags)
```

Comment:
**non-blocking nitpick/question:** Perhaps we would want to have these params live in separate lines instead of all in one line except for the last one (overflown)?

Just a thought since we have a few different ways we handle calls will many/lengthy parameters in them ([one example here](https://github.com/forem/forem/blob/9250b553d0f9e5b2552d64c13248f18744456736/app/controllers/feedback_messages_controller.rb#L14)). This is a personal opinion on readability so very non-blocking comment here 🙂

Reply:
done, thank you!